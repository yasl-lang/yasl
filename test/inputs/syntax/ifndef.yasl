
ifdef !A {
    echo 'A'
} elseifdef !B {
    echo 'B'
} else {
    echo 'neither'
}

ifdef !collections {
    echo 'collections'
} else {
    echo undef
}

ifdef !asddas {
    echo 'asddas'
} elseifdef !math {
    echo 'math'
} else {
    echo undef
}

const B = 'B'

fn f(A) {
    ifdef !A {
        echo 'A'
    } elseifdef !B {
        echo 'B'
    } else {
        echo 'neither'
    }

    ifdef !collections {
        echo 'collections'
    } else {
        echo undef
    }

    ifdef !B {
        echo 'B'
    } elseifdef !math {
        echo 'math'
    } else {
        echo undef
    }
}

f()
