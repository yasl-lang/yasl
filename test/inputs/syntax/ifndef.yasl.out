A
undef
asddas
neither
undef
undef
